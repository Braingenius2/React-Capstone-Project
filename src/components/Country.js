import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import { FiArrowRightCircle } from 'react-icons/fi';

const Country = ({
  flag, name, covidCases, index, columnIndex,
}) => {
  const isEvenRow = Math.floor(index / 2) % 2 === 0;
  const isLightPink = (isEvenRow && columnIndex === 0) || (!isEvenRow && columnIndex === 1);
  const backgroundColor = isLightPink ? 'bg-pink-600' : 'bg-pink-500';

  const [isHovered, setIsHovered] = useState(false);

  const handleMouseEnter = () => {
    setIsHovered(true);
  };

  const handleMouseLeave = () => {
    setIsHovered(false);
  };
  const countryName = name.replace(/\s/g, '-');

  return (
    <NavLink
      to={`${countryName}`}
      className={`p-4 ${backgroundColor} country-item`}
      onMouseEnter={handleMouseEnter}
      onMouseLeave={handleMouseLeave}
    >
      <div>
        <img
          src={flag}
          alt={`flag of ${countryName}`}
          style={{
            filter: isHovered ? 'opacity(1)' : 'opacity(0.7)',
          }}
        />
      </div>
      <div>
        <h3>{name}</h3>
        <p>
          {covidCases}
          {' '}
          Covid Cases
        </p>
      </div>
      <div>
        <FiArrowRightCircle />
      </div>
    </NavLink>
  );
};

Country.propTypes = {
  flag: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  covidCases: PropTypes.string.isRequired,
  index: PropTypes.number.isRequired,
  columnIndex: PropTypes.number.isRequired,
};

export default Country;
