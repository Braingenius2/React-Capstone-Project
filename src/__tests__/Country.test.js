import '@testing-library/jest-dom/extend-expect';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Country from '../components/Country';

describe('The Country component', () => {
  test('should render correctly into the DOM', () => {
    const { container } = render(
      <BrowserRouter>
        <Country image="https://disease.sh/assets/img/flags/af.png" name="Afghanistan" cases="224,317" />
      </BrowserRouter>,
    );
    expect(container).toMatchSnapshot();
  });

  test('should render the text: Algeria and the number of cases', () => {
    render(
      <BrowserRouter>
        <Country image="https://disease.sh/assets/img/flags/dz.png" name="Algeria" cases="271,852" />
      </BrowserRouter>,
    );
    const countryTitle = screen.getByText(/Algeria/i);
    const countryCases = screen.getByText(/271,852 cases/i);

    expect(countryTitle).toBeInTheDocument();
    expect(countryCases).toBeInTheDocument();
  });
});
