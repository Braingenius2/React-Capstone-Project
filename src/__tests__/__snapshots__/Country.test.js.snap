// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Country component should render correctly into the DOM 1`] = `undefined`;
